register.lic

```diff
@@ -0,0 +1,76 @@
+=begin
+  Search a deed register for the specified crafting material.
+  Output all lines that match.
+=end
+
+custom_require.call(%w[common common-items drinfomon])
+
+class Register
+  include DRC
+  include DRCI
+
+  def initialize
+    settings = get_settings
+	@search_results = []
+
+    arg_definitions =
+      [
+        [
+          { name: 'query', regex: /^[A-z\s\-\']+$/i, description: 'crafting material to search for, use quotes for multiple words' }
+        ]
+      ]
+
+    args = parse_args(arg_definitions)
+
+    if args.query
+      stow_hands
+	  if get_item('deed register', get_settings.crafting_container)
+	    fput("turn my register to contents")
+        search(args.query)
+	  else
+	    echo "Can\'t find your deed register."
+		exit
+	  end
+      stow_hands
+	end
+	respond('   ' + "Results:")
+	if @search_results.any?
+	  display_results(@search_results)
+	else
+	  respond('   ' + "No matches found.")
+	end
+  end
+
+  def display_results(results)
+    results.each do |data|
+      respond("#{data}")
+    end
+  end
+
+  def search(query)
+    contents = []
+    fput("read my register")
```

Comment:
Would you be up for making this a `bput`?

Reply:
Same answer as above.

Reply:
Changing this one makes the script very laggy when reading the register.

Reply:
Never mind.  It was removing the pause that made it slow.

Overall though, the bputs do seem to slow down the script just a tad.

Reply:
See comment above.